plugins/cue_points/quiz/lib/kQuizManager.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * @package plugins.quiz
+ * @subpackage quiz
+ */
+class kQuizManager implements kObjectChangedEventConsumer
+{
+	const EVENT_TYPE = 'eventType';
+	const KUSER_ID = 'kuserId';
+	const ENTRY_ID = 'entryId';
+	const PARTNER_ID = 'partnerId';
+	const EVENT_TIME = 'eventTime';
+	const VIRTUAL_EVENT_ID = 'virtualEventId';
+	const VERSION = 'version';
+	const ANSWER_IDS = 'answerIds';
+	const SCORE = 'score';
+	const NUM_OF_CORRECT_ANSWERS = 'numOfCorrectAnswers';
+	const NUM_OF_QUESTIONS = 'numOfQuestions';
+	const NUM_OF_RELEVANT_QUESTIONS = 'numOfRelevnatQuestions';
+	const CALCULATED_SCORE = 'calculatedScore';
+	
+	const MAP_NAME = 'internal_analytics_host';
+	const HEADER_HOST = 'Host';
+	const HEADER_IDENTIFIER = 'X-Forwarded-For';
+	const HEADER_AGENT = 'User-Agent';
+	const HEADER_AGENT_TYPE = 'HTTP_USER_AGENT';
+	
+	/**
+	 * @param BaseObject $object
+	 * @param array $modifiedColumns
+	 * @return bool
+	 */
+	public function objectChanged (BaseObject $object, array $modifiedColumns)
+	{
+		$quizEventContent = array();
+		if (self::wasStatusChanged($object, $modifiedColumns))
+		{
+			$quizEventContent = self::generateQuizEventContent($object);
+		}
+		$uri = '/api_v3/index.php?' . http_build_query($quizEventContent, '', '&');
+		
+		self::sendBeacon($uri);
+	}
+	
+	protected static function sendBeacon (string $uri)
```

Comment:
move the duplicated code to function which will be used for both of the deacon existing types